pkg/login/social/social.go

```diff
@@ -79,7 +94,7 @@ func newSocialBase(name string, config *oauth2.Config, info *setting.OAuthInfo)
 	}
 }
 
-func NewOAuthService(cfg *setting.Cfg) {
+func (ss *Service) NewOAuthService(cfg *setting.Cfg) {
```

Comment:
Think you can move all of this into the `Init` method now. 

The `setting.OAuthService` would be nice to move into the `Service` struct as a field and move the `setting.OAuther`. `setting.OAuthInfo` and `setting.OAuthService` into the social package. WDYT?

Reply:
I can do that